src/coreclr/jit/runtimelookup.cpp

```diff
@@ -0,0 +1,434 @@
+                    // 80% chance we pass nullcheck
+                    nullcheckBb->inheritWeightPercentage(sizeCheckBb, 80);
+                    // 64% (0.8 * 0.8) chance we pass both nullcheck and sizecheck
+                    fastPathBb->inheritWeightPercentage(nullcheckBb, 80);
+                    // 100-64=36% chance we fail either nullcheck or sizecheck
+                    fallbackBb->inheritWeightPercentage(sizeCheckBb, 36);
+                }
+                else
+                {
+                    nullcheckBb->inheritWeight(prevBb);
+                    // 80% chance we pass nullcheck
+                    fastPathBb->inheritWeightPercentage(nullcheckBb, 80);
+                    // 20% chance we fail nullcheck (TODO: Consider making it cold (0%))
+                    fallbackBb->inheritWeightPercentage(nullcheckBb, 20);
+                }
+
+                //
+                // Update loop info if loop table is known to be valid
+                //
+                if (optLoopTableValid && prevBb->bbNatLoopNum != BasicBlock::NOT_IN_LOOP)
+                {
+                    nullcheckBb->bbNatLoopNum = prevBb->bbNatLoopNum;
+                    fastPathBb->bbNatLoopNum  = prevBb->bbNatLoopNum;
+                    fallbackBb->bbNatLoopNum  = prevBb->bbNatLoopNum;
+                    if (needsSizeCheck)
+                    {
+                        sizeCheckBb->bbNatLoopNum = prevBb->bbNatLoopNum;
+                    }
+                    // Update lpBottom after block split
+                    if (optLoopTable[prevBb->bbNatLoopNum].lpBottom == prevBb)
+                    {
+                        optLoopTable[prevBb->bbNatLoopNum].lpBottom = block;
+                    }
+                }
+
+                // All blocks are expected to be in the same EH region
+                assert(BasicBlock::sameEHRegion(prevBb, block));
+                assert(BasicBlock::sameEHRegion(prevBb, nullcheckBb));
+                assert(BasicBlock::sameEHRegion(prevBb, fastPathBb));
+                if (needsSizeCheck)
+                {
+                    assert(BasicBlock::sameEHRegion(prevBb, sizeCheckBb));
+                }
+
+                // Scan current block again, the current call will be ignored because of ClearExpRuntimeLookup.
+                // We don't try to re-use expansions for the same lookups in the current block here - CSE is responsible
+                // for that
+                result = PhaseStatus::MODIFIED_EVERYTHING;
+            }
+        }
+    }
+
+    if (result == PhaseStatus::MODIFIED_EVERYTHING)
+    {
+        if (opts.OptimizationEnabled())
+        {
+            fgReorderBlocks(/* useProfileData */ false);
+            fgUpdateChangedFlowGraph(FlowGraphUpdates::COMPUTE_BASICS);
+        }
+
+#ifdef DEBUG
+        if (verbose)
+        {
+            printf("\n*************** After fgExpandRuntimeLookups()\n");
+            fgDispBasicBlocks(true);
+        }
+#endif
```

Comment:
Also think this should be necessary, post-phase handling is going to print the IR if you return `PhaseStatus::MODIFIED_EVERYTHING`.